### PR TITLE
readdlm file input checking

### LINKED ISSUE #11484: better error message for readdlm on directories
Hi, 

It took me a few minutes of headache to understand why 

``` julia
readdlm("I-typed-a-directory-instead-of-a-file-path")
```

gives 

``` julia
memory mapping failed: No such device
while loading In[37], in expression starting on line 1

 in mmap at mmap.jl:35
 in mmap_array at mmap.jl:110
 in mmap_array at mmap.jl:4
 in anonymous at datafmt.jl:64
 in open at /usr/bin/../lib/julia/sys.so
 in as_mmap at datafmt.jl:63
 in readdlm_auto at datafmt.jl:55
 in readdlm at datafmt.jl:49
 in readdlm at datafmt.jl:46
```

Probably a better error handling in these cases would help.

Davide

### PR DESCRIPTION
**Pull request by stillyslalom**
Check that input string path is a file (resolves #11484).

### CODE REVIEW

File: base/datafmt.jl [hunk 1]
```diff
@@ -119,6 +119,7 @@ readdlm_auto(input::Vector{UInt8}, dlm::Char, T::Type, eol::Char, auto::Bool; op
 readdlm_auto(input::IO, dlm::Char, T::Type, eol::Char, auto::Bool; opts...) =
     readdlm_string(read(input, String), dlm, T, eol, auto, val_opts(opts))
 function readdlm_auto(input::AbstractString, dlm::Char, T::Type, eol::Char, auto::Bool; opts...)
+    isfile(input) || error("Cannot read \'$input\': not a file")
```
Comment (Sacha0) [inline base/datafmt.jl]: Perhaps `throw(SystemError("..."))` rather than `error("...")`?

**Comment by Sacha0** [comment]
Thanks @stillyslalom! :) Perhaps add a test for the error? Best!

File: base/datafmt.jl (hunk 1 above)
Comment (stillyslalom) [inline base/datafmt.jl]: I think `ArgumentError` might be a better fit, since the error's due to bad input

File: test/datafmt.jl [hunk 2]
```diff
@@ -204,7 +204,7 @@ let i18n_data = ["Origin (English)", "Name (English)", "Origin (Native)", "Name
         "Russia", "Mikhail Gorbachev", "Россия", "Михаил Горбачёв",
         "Russia", "Boris Grebenshchikov", "Россия", "Борис Гребенщиков",
         "Slovenia", "\"Frane \"\"Jezek\"\" Milcinski", "Slovenija", "Frane Milčinski - Ježek",
-        "Syracuse (Sicily)", "Archimedes", "Συρακούσα", "Ἀρχιμήδης",
+        "Syracuse (Sicily)", "Archimedes", "Συρακούσα", "Ἀρχιμήδης",
```
Comment (stillyslalom) [inline test/datafmt.jl]: I'm not sure why these were diffed - I didn't modify it at all

Comment (quinnj) [inline test/datafmt.jl]: Maybe an encoding change from editor settings?

File: base/datafmt.jl (hunk 1 above)
Comment (Sacha0) [inline base/datafmt.jl]: `ArgumentError` also seems reasonable :). I suggested `SystemError` on the basis of `open`'s behavior in the same case:
```julia
julia> open("foo")
ERROR: SystemError: opening file foo: No such file or directory
Stacktrace:
 [1] #systemerror#44 at ./error.jl:77 [inlined]
 [2] systemerror at ./error.jl:77 [inlined]
 [3] open(::String, ::Bool, ::Bool, ::Bool, ::Bool, ::Bool) at ./iostream.jl:104
 [4] open(::String) at ./iostream.jl:113
```

File: test/datafmt.jl (hunk 2 above)
Comment (stillyslalom) [inline test/datafmt.jl]: I made the change from the Github file editor, which I assume takes its encoding from the host browser? It seems to be UTF8.

File: base/datafmt.jl (hunk 1 above)
Comment (stillyslalom) [inline base/datafmt.jl]: The issue was originally filed b/c someone tried to `readdlm` an existing directory. None of the Libc errors seem to really fit.

File: test/datafmt.jl (hunk 2 above)
Comment (StefanKarpinski) [inline test/datafmt.jl]: Probably Unicode normalization change – both strings have accents which have composed and decomposed forms. Doesn't really matter which we use, so this is a fairly harmless change.

**Comment by stillyslalom** [comment]
Anything preventing this from being merged?

**Pull request merged**